### PR TITLE
refactor: improve ledger performance. fixes #330

### LINKED ISSUE #330: InsufficientFundsError on hold account
I'm stress testing the wallet by submitting multiple (2-3) parallel payments and sometimes I get this strange exception from the ledger when the plugin-bells tries to submit the fulfillment.

```
{"id":"InsufficientFundsError","message":"Sender has insufficient funds.","owner":"hold"}
```

### PR DESCRIPTION
**Pull request by vhpoet**
See the discussion in https://github.com/interledger/five-bells-ledger/issues/330.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -91,6 +91,7 @@
     "sinon-chai": "^2.7.0",
     "spec-xunit-file": "0.0.1-3",
     "supertest": "^1.1.0",
+    "supertest-as-promised": "^4.0.0",
```
Comment (justmoon) [inline package.json]: You don't need this. Just use `this.request()`...`.end()`  - That will return a promise.

File: test/fulfillmentSpec.js [hunk 2]
```diff
@@ -348,6 +349,12 @@ describe('PUT /fulfillment', function () {
     })
 
   it('should not double spend when transfer is executed multiple times', function * () {
+    // Pre-fund hold account to allow double spending of source transfer
+    // This could happen if, for ex, 2 txn are escrowed, then first is executed twice.
```
Comment (justmoon) [inline test/fulfillmentSpec.js]: It seems like we should fix the first transaction being executed twice, then we wouldn't need this hack?

Comment (vhpoet) [inline test/fulfillmentSpec.js]: @justmoon @emschwartz I'm not very familiar with the code, does anyone want to look at this?

Comment (MatthewPhinney) [inline test/fulfillmentSpec.js]: Not sure what you mean by "transaction" (transfer? database transaction?). We had a separate issue we fixed with the notary, where it was sending two fulfillments per transfer. If we only get one fulfillment per transfer, this won't happen, but I think it's good to have a defensive test to confirm the ledger behaves properly, even in that case.

**Comment by emschwartz** [comment]
Should we close this or update and merge it? @vhpoet @justmoon

**Comment by justmoon** [comment]
@emschwartz Just went in and fixed the PR. Major changes, can you review please?

**Comment by michielbdejong** [comment]
See also https://github.com/interledgerjs/five-bells-shared/pull/68 which already introduced retry logic at the `five-bells-shared` level

**Comment by michielbdejong** [comment]
I'll try to merge the two as part of https://github.com/interledgerjs/five-bells-ledger/pull/408

**Pull request merged**